### ISSUE TITLE
didn't block youtube

### ISSUE THREAD

**Issue by rub1e** [matchu/Strict-Workflow#52]
on my chromebook, I tried to access youtube during a work phase and it initially seemed like it had been blocked, but the video actually loaded invisibly and I could hear audio before it seemed to crash. But when I reloaded the page, the video worked fine, and youtube worked fine in that tab, even during a work phase.

Great tool otherwise!

**Comment by matchu** [comment]
Yeah :/ I'm not sure how to handle this. When blocking a page, we only
cover it up rather than actually replacing the page's contents, because, if
a user starts a work phase while filling out a form, I don't want them to
lose all their form input. Similarly, if the user is doing something
important in a Flash object, I don't want them to lose their progress—but,
if I leave the Flash object intact, then it's free to play whatever audio
it wants.

I guess there are a few options here.

* We could leave the behavior as-is.
* We could just tear down the page contents anyway, and expect users to
deal with it.
* We could provide an option for whether you want to block pages by
covering them up or replacing them entirely.
* We could provide a one-off way to remove the contents of a *particular*
noisy tab.
* We could provide special behavior for websites like Youtube, like sending
a Pause message to the player when the tab is blocked.
  * This option seems interesting to me, but I'm hesitant about the
precedent it sets. I don't wanna write pause behavior for every website in
the world…

**Comment by matchu** [comment]
Yeah :/ I'm not sure how to handle this. When blocking a page, we only cover it up rather than actually replacing the page's contents, because, if a user starts a work phase while filling out a form, I don't want them to lose all their form input. Similarly, if the user is doing something important in a Flash object, I don't want them to lose their progress—but, if I leave the Flash object intact, then it's free to play whatever audio it wants.

I guess there are a few options here.

* We could leave the behavior as-is.
* We could just tear down the page contents anyway, and expect users to
deal with it.
* We could provide an option for whether you want to block pages by
covering them up or replacing them entirely.
* We could provide a one-off way to remove the contents of a *particular*
noisy tab.
* We could provide special behavior for websites like Youtube, like sending
a Pause message to the player when the tab is blocked.
  * This option seems interesting to me, but I'm hesitant about the
precedent it sets. I don't wanna write pause behavior for every website in
the world…

**Comment by rub1e** [comment]
That last option does seem interesting, and I take your point about the precedent, but you could start with just the big ones - youtube, and maybe vimeo. But I guess it's actually quite tricky to second-guess what users would want:  

* I'd be quite annoyed if I were in the middle of a video and it continued to play during a work phase with no way for me to pause it  
* But then that goes against the entire principle of the work timer if it's down to me to pause it
* But then it would also be frustrating if you just muted that tab and I lost my place in the video I was watching
* But then...

My vote would tend towards the pause option just for the most likely sites - i.e. youtube.

Shall I close this issue, or leave it open for others to comment on?

**Comment by rub1e** [comment]
Just had a thought - since the work phase is user-initiated, couldn't you just have an option to mute audio on blacklisted sites and warn users to pause any videos/songs they're watching before they hit the work timer, because otherwise they'll continue to play on mute and they'll lose their place?

And generally you could warn people to finish activities with user input like forms before hitting the work timer?

**Comment by matchu** [comment]
Yeah, warnings are a thought—they were in a draft of a new UX at one point,
but that hasn't gotten as far as I'd like…
